easyspin/unitconvert.m

```diff
@@ -36,59 +36,53 @@
 
-    case 'cm^-1->eV'
-        out = value.*100*clight*planck/evolt;
-    case 'cm^-1->K'
-        out = value.*100*clight*planck/boltzm;
-    case 'cm^-1->mT'
-        out = value./g*(planck/bmagn/1e-3)*100*clight;
-    case 'cm^-1->MHz'
-        out = value.*100*clight/1e6;
+unitconv = {
+  "cm^-1->eV",    @(value) value.*100*clight*planck/evolt;
+  "cm^-1->K",     @(value) value.*100*clight*planck/boltzm;
+  "cm^-1->mT",    @(value) value./g*(planck/bmagn/1e-3)*100*clight;
+  "cm^-1->MHz",   @(value) value.*100*clight/1e6;
 
+  "eV->cm^-1",    @(value) value.*evolt/100/clight/planck;
+  "eV->K",        @(value) value.*evolt/boltzm;
+  "eV->mT",       @(value) value./g/bmagn/1e-3*evolt;
+  "eV->MHz",      @(value) value.*evolt/planck/1e6;
 
-    case 'eV->cm^-1'
-        out = value.*evolt/100/clight/planck;
-    case 'eV->K'
-        out = value.*evolt/boltzm;
-    case 'eV->mT'
-        out = value./g/bmagn/1e-3*evolt;
-    case 'eV->MHz'
-        out = value.*evolt/planck/1e6;
+  "K->cm^-1",     @(value) value.*boltzm/100/clight/planck;
+  "K->eV",        @(value) value.*boltzm/evolt;
+  "K->mT",        @(value) value./g/bmagn/1e-3*boltzm;
+  "K->MHz",       @(value) value.*boltzm/planck/1e6;
 
+  "mT->cm^-1",    @(value) value.*g/(planck/bmagn/1e-3)/100/clight;
+  "mT->eV",       @(value) value.*g*bmagn*1e-3/evolt;
+  "mT->K",        @(value) value.*g*bmagn*1e-3/boltzm;
+  "mT->MHz",      @(value) value.*g*(1e-3*bmagn/planck/1e6);
 
-    case 'K->cm^-1'
-        out = value.*boltzm/100/clight/planck;
-    case 'K->eV'
-        out = value.*boltzm/evolt;
-    case 'K->mT'
-        out = value./g/bmagn/1e-3*boltzm;
-    case 'K->MHz'
-        out = value.*boltzm/planck/1e6;
+  "MHz->cm^-1",   @(value) value.*1e6/100/clight;
+  "MHz->eV",      @(value) value.*1e6*planck/evolt;
+  "MHz->K",       @(value) value.*1e6*planck/boltzm;
+  "MHz->mT",      @(value) value./g*(planck/bmagn/1e-3)*1e6
+  };
 
+% List of possible unit conversion as string array
+unames = string(unitconv(:,1));
 
-    case 'mT->cm^-1'
-        out = value.*g/(planck/bmagn/1e-3)/100/clight;
-    case 'mT->eV'
-        out = value.*g*bmagn*1e-3/evolt;
-    case 'mT->K'
-        out = value.*g*bmagn*1e-3/boltzm;
-    case 'mT->MHz'
-        out = value.*g*(1e-3*bmagn/planck/1e6);
+% Test if input unit conversion exists (case-sensitive)
+unitMatch = strcmp(unames,units);
 
-
-    case 'MHz->cm^-1'
-        out = value.*1e6/100/clight;
-    case 'MHz->eV'
-        out = value.*1e6*planck/evolt;
-    case 'MHz->K'
-        out = value.*1e6*planck/boltzm;
-    case 'MHz->mT'
-        out = value./g*(planck/bmagn/1e-3)*1e6;
-        
-
-    otherwise
-        error('Unknown unit conversion specified ')
-
-end
+if any(unitMatch)
+  % Return unit conversion
+  out = unitconv{unitMatch,2}(value);
+else
+  % Throw errors depending on unit input
+  % Test if input unit conversion exists (case-insensitive)
+  unitMatchI = strcmpi(unames,units);
+  if any(unitMatch)
```

Comment:
Shouldn't this be `any(unitMatchI)`?

Reply:
Indeed. I found another small error in the error handling. With the new commit it should be solved.

Reply:
I just realised that the minimum required Matlab version for arrays of strings is R2018b (c.f. #160 ). It would also be possible to use a structure instead. I am not such a fan of structure though.